packages/cdktf-cli/bin/cdktf.ts

```diff
@@ -27,6 +28,48 @@ if (semver.lt(process.version, "10.12.0")) {
   process.exit(1);
 }
 
+const customCompletion = function (
+  _current: string,
+  argv: any,
+  completionsFilter: (
+    done?: (err: any, defaultCompletions: string[]) => void
+  ) => void,
+  done: (completions: string[]) => void
+) {
+  // The completion function gets called when bash completion is invoked and
+  // we have very little context there. So when debugging via logs use e.g.
+  // fs.appendFileSync("/Users/ansgar/cdktf-completions.log", `hello log\n`);
+
+  // complete positional arg for stack for diff / deploy (and their aliases plan / apply)
+  if (
+    argv._.includes("diff") ||
+    argv._.includes("deploy") ||
+    argv._.includes("plan") ||
+    argv._.includes("apply")
+  ) {
+    completionsFilter(async (_err, defaults) => {
+      const stacks: string[] = [];
+      try {
+        const manifest = await readCDKTFManifest();
```

Comment:
The end result would be the same, but would it be better to check for the existence of the manifest instead?

Reply:
It could also fail if the Manifest would not be readable somehow (e.g. file permissions) or if it would be invalid json (aka be corrupted), so I opted for the safe way here.